src/modules/IscsiClientLib.rb

```diff
@@ -748,44 +763,46 @@ def deleteRecord
       ret
     end
 
-    # get (manual/onboot) status of target connecting
+    # Get info about current iSCSI node
+    #
+    # return [Hash]    stdout of 'iscsiadm -m node -I <iface> -T <target> -p <ip>'
+    #                  converted to a hash
+    def getCurrentNodeValues
+      node_val = {}
+
+      ret = SCR.Execute(path(".target.bash_output"),
+                        GetAdmCmd("-m node -I #{@currentRecord[2]||"default"} -T #{@currentRecord[1]||""} -p #{@currentRecord[0]||""}"))
+      return node_val unless (ret["exit"] || -1) == 0
+
+      node_val = nodeInfoToMap(ret["stdout"] || "")
```

Comment:
I do not see reason why to assign to node_val. Maybe simply use above `return {} if ret["exit"] != 0` and then just `nodeInfoToMap(ret["stdout"] || "")`